src/xb-main.c

```diff
@@ -90,46 +89,14 @@ server_send_response (SoupMessage *message,
 
 static char *
 resolve_flatpak_path (const char *path,
-                      const char *app_id)
+                      const char *app_path)
 {
   g_autofree char *real_path = g_file_read_link (path, NULL);
   if (!real_path)
     real_path = g_strdup (path);
 
-  g_autoptr(FlatpakInstalledRef) app_ref = NULL;
-
-  g_autoptr(FlatpakInstallation) user_install =
-    flatpak_installation_new_user (NULL, NULL);
-
-  if (user_install)
-    {
-      app_ref =
-        flatpak_installation_get_current_installed_app (user_install,
-                                                        app_id,
-                                                        NULL, NULL);
-    }
-
-  if (!app_ref)
-    {
-      g_autoptr(FlatpakInstallation) system_install =
-        flatpak_installation_new_system (NULL, NULL);
-
-      if (system_install)
-        {
-          app_ref =
-            flatpak_installation_get_current_installed_app (system_install,
-                                                            app_id,
-                                                            NULL, NULL);
-        }
-    }
-
-  if (!app_ref)
-    return NULL;
-
-  const char *deploy_dir = flatpak_installed_ref_get_deploy_dir (app_ref);
   const char *relative_path = real_path + strlen ("/app");
-
-  return g_build_filename (deploy_dir, "files", relative_path, NULL);
+  return g_build_filename (app_path, relative_path, NULL);
```

Comment:
Is there any reason you can't do this resolution entirely within eos-knowledge-lib? It's a bit bizarre to me...